docs/ui-components/code.md

```diff
@@ -156,8 +156,11 @@ In order to use a local version of the mermaid library instead of one provided b
 mermaid:
   # To load mermaid from a local file use the `path` key to specify the location of the library instead; e.g.
   path: "/assets/js/mermaid.min.js"
+  # note: you must still specify a version! Behaviour is different for versions < 10 and >= 10
 ```
 
+For mermaid versions `>=10`, this file is imported directly as an ESM module (rather than as a `<script>` tag); users should use the `mermaid.esm.min.js` file. In contrast, for mermaid versions `<=10`, this file is loaded as a script tag; it should be a standalone CJS file (ex `mermaid.min.js`).
+
```

Comment:
```suggestion
```

Reply:
You wrote:

> There is no single JS file that bundles the entire library into one minified file; the `mermaid.esm.min.mjs` file still has relative imports off of the CDN.

Sorry, but I've become quite confused. Isn't that saying that it is _not_ possible to use the ESM file as a local library? That's inconsistent with "users should use the `mermaid.esm.min.js` file".

And CDN users don't specify a `path`, so what difference does it make for them whether the specified version is imported as an ESM module or as a standalone file?

I have no background knowledge of ESM modules or mermaid. I suggested removing that item from the docs because part of it seemed inconsistent, and the rest irrelevant. If that's due to my misunderstanding, please reject the suggestion and clarify how to read it.

Reply:
Ah, I could have been more clear - that's on me!

> You wrote:
> 
> > There is no single JS file that bundles the entire library into one minified file; the `mermaid.esm.min.mjs` file still has relative imports off of the CDN.
> 
> Sorry, but I've become quite confused. Isn't that saying that it is _not_ possible to use the ESM file as a local library? That's inconsistent with "users should use the `mermaid.esm.min.js` file".

Broadly, the library is ESM-only. It is unclear that mermaid will ever bundle non-ESM versions of the library again. I thought that they would still release versions of `mermaid.min.js`, but it seems like that's not the case. So, this means that `v10+` local users will have to use the ESM version of the library. So, I've included that as the instruction.

*Completely separately*, the ESM bundle for `mermaid.min.js` is not standalone, since it references other modules in the folder. Currently, users can copy the *entire* `dist` directory (maintaining the relative file paths) and use an ESM bundle. I believe this will be resolved in an upcoming release (hence this being the "temporary change").

> And CDN users don't specify a path, so what difference does it make for them whether the specified version is imported as an ESM module or as a standalone file?

I agree; but, this is in the subsection of the docs on using a local library. I think it is relevant for those users.

> I have no background knowledge of ESM modules or mermaid. I suggested removing that item from the docs because part of it seemed inconsistent, and the rest irrelevant. If that's due to my misunderstanding, please reject the suggestion and clarify how to read it.

My intention was to say that the file specified by `path` is loaded in two different ways, depending on the version:

- for versions `<10`, as a "vanilla"/CJS module
- for versions `>=10`, as an ESM module

Is there a way I could make that clearer?